include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -155,6 +155,21 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         .wait();
 }
 
+template <typename Iterator1, typename Iterator2, std::enable_if_t<std::is_same_v<Iterator1, Iterator2>, int> = 0>
```

Comment:
I would like to notice that for example a case `const Iterator1 `and `Iterator2` should work as well here.
Because it is a typical case when input iterator is const, output is non-const (of course).

Reply:
Already fixed.